examples/run/cuda/seeding_comparison_example_cuda.cpp

```diff
@@ -0,0 +1,456 @@
+            evt_data.generate_truth_candidates(sg, host_mr,
+                                               0.5 / unit<float>::GeV);
+        traccc::edm::spacepoint_collection::host spacepoints_per_event{host_mr};
+        traccc::measurement_collection_types::host measurements_per_event{
+            &host_mr};
+        traccc::io::read_spacepoints(
+            spacepoints_per_event, measurements_per_event, event,
+            input_opts.directory,
+            (input_opts.use_acts_geom_source ? &detector : nullptr),
+            input_opts.format);
+
+        std::map<measurement, std::size_t>
+            measurement_to_evt_data_measurement_map;
+        std::size_t num_unmatched_measurements = 0;
+
+        for (std::size_t i = 0; i < measurements_per_event.size(); ++i) {
+            bool matched = false;
+
+            for (const auto& kv : evt_data.m_measurement_map) {
+                if (kv.second == measurements_per_event.at(i)) {
+                    measurement_to_evt_data_measurement_map
+                        [measurements_per_event.at(i)] = kv.first;
+                    matched = true;
+                    break;
+                }
+            }
+
+            if (!matched) {
+                num_unmatched_measurements++;
+            }
+        }
+
+        if (num_unmatched_measurements > 0) {
+            TRACCC_WARNING("At least " << num_unmatched_measurements
+                                       << " measurements are unmatched");
+        } else {
+            TRACCC_INFO("All measurements are matched; nice!");
+        }
+
+        std::vector<output_data_point> output_data_points;
+        std::map<std::tuple<measurement, measurement, measurement>, std::size_t>
+            seed_to_output_data_point_idx_map;
+
+        // === TRUTH SEEDING ===
+        std::size_t num_truth_tracks = 0;
+
+        {
+            std::size_t num_untrackable_particles = 0;
+
+            // Prepare truth seeds
+            for (std::size_t i_trk = 0; i_trk < truth_track_candidates.size();
+                 i_trk++) {
+                if (truth_track_candidates.at(i_trk).items.size() < 3) {
+                    num_untrackable_particles++;
+                    continue;
+                }
+
+                const measurement& bot_meas =
+                    truth_track_candidates.at(i_trk).items.at(0);
+                const measurement& mid_meas =
+                    truth_track_candidates.at(i_trk).items.at(1);
+                const measurement& top_meas =
+                    truth_track_candidates.at(i_trk).items.at(2);
+
+                const std::map<particle, std::size_t>& bot_ptc_map =
+                    evt_data.m_meas_to_ptc_map.at(bot_meas);
+                const std::map<particle, std::size_t>& mid_ptc_map =
+                    evt_data.m_meas_to_ptc_map.at(mid_meas);
+                const std::map<particle, std::size_t>& top_ptc_map =
+                    evt_data.m_meas_to_ptc_map.at(top_meas);
```

Comment:
@beomki-yeo I've noticed that if I get the bottom, middle, and top spacepoint positions like this, I get non-sensical $r$ progression on $79.2\%$ of seeds, i.e. the bottom $r$ isn't lower than the middle $r$ or the middle $r$ isn't lower than the top $r$. Is this expected? Do I need to sort the track candidate states?

Reply:
It is not expected and the track candidates do not have to be sorted.

How did you get the spacepoint positions from `bot_meas`, `mid_meas` and `top_meas`?

Reply:
After going through `event_data.cpp`, it is possible that we may need to sort the track candidate if `csv_measurements` is not sorted in the timing order:
https://github.com/acts-project/traccc/blob/6fee5ca3638f7a4572719a670047a48f05b4907c/performance/src/utils/event_data.cpp#L230

Reply:
I found that `measurement.csv` of geant4 odd data does not have truth timing information (all of them are set to zero).
We will have to find the truth time from the corresponding hit object in `hits.csv`, which has truth timing information, via `measurement-simhit-map.csv`

Then you can run sorting on measurement (track candidate) vectors w.r.t. truth timing